src/generators/mandoc/index.mjs

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+import { optionToMandoc, envToMandoc } from './utils/converter.mjs';
+import { writeFile, readFile } from 'node:fs/promises';
+import { join } from 'node:path';
+
+/**
+ * This generator generates a mandoc version of the API docs
+ *
+ * @typedef {Array<ApiDocMetadataEntry>} Input
+ *
+ * @type {import('../types.d.ts').GeneratorMetadata<Input, string>}
+ */
+export default {
+  name: 'mandoc',
+
+  version: '1.0.0',
+
+  description: 'Generates the `node.1` file.',
+
+  dependsOn: 'ast',
+
+  async generate(input, options) {
+    // Find the appropriate headers
+    const optionsStart = input.findIndex(({ slug }) => slug === 'options');
+    const environmentStart = input.findIndex(
+      ({ slug }) => slug === 'environment-variables-1'
+    );
+
+    // Generate the option mandoc
+    let optionsOutput = '';
+    for (let i = optionsStart + 1; i < environmentStart; i++) {
```

Comment:
Can you use forEach here? In general we prefer usage of builtins here. Also heading is supposed to exist on all nodes I think.

Reply:
We aren't starting at index `0`, so using a for-loop makes more sense. I could switch it, it just would be less efficient.

Reply:
Also, it has to be a `for` loop for `envOutput`, as we are exiting early in some cases. Wouldn't it make more sense to keep them both for consistency? If you disagree, I'll change it.

Reply:
IDK, I rather usually use builtins, always easier to read